library/kani_macros/src/sysroot/contracts.rs

```diff
@@ -809,6 +904,40 @@ fn requires_ensures_main(attr: TokenStream, item: TokenStream, is_requires: bool
     output.into()
 }
 
+fn pats_to_idents<P>(
+    sig: &mut syn::punctuated::Punctuated<syn::FnArg, P>,
+) -> impl Iterator<Item = Ident> + '_ {
+    sig.iter_mut().enumerate().map(|(i, arg)| match arg {
+        syn::FnArg::Receiver(_) => Ident::new("self", Span::call_site()),
```

Comment:
Should `"self"` be replaced with [`SelfValue`](https://docs.rs/syn/latest/syn/token/struct.SelfValue.html)?

Reply:
Hmm, well then the function would have to return something like `Either<Ident, SelfValue>` because I don't think there's a canonical way to turn a `SelfValue` into an `Ident`

Reply:
I see. It might make sense to use it in `visit_path` below (see my next comment).

Reply:
I'm using it now via `Token!`, maybe that's what you meant?

Reply:
Yes, that's what I meant.